client/app/pages/queries/hooks/useAutoLimitFlags.js

```diff
@@ -0,0 +1,24 @@
+import { useCallback, useState } from "react";
+import localOptions from "@/lib/localOptions";
+import { get, extend } from "lodash";
+
+function isAutoLimitAvailable(dataSource) {
+  return get(dataSource, "apply_auto_limit", false);
+}
+
+export default function useAutoLimitFlags(dataSource, query, setQuery) {
+  const isAvailable = isAutoLimitAvailable(dataSource);
+  const [isChecked, setIsChecked] = useState(localOptions.get("applyAutoLimit", true));
+  query.options.applyAutoLimit = isAvailable && isChecked;
```

Comment:
Is this to guarantee the `applyAutoLimit` is set to false when not available? If so, how about we put this line into a

```jsx
useEffect(() => {
  query.options.applyAutoLimit = isAvailable && isChecked;
}, [isAvailable, isChecked]);
```
? This way it will update that value only when one the vars change and not on every render.

Reply:
Maybe I'm missing something, but it's a simple operation between two boolean values which most of the time yields the same value (so no change in rendering). Why complicate the code with a `useEffect`?

Reply:
I thought it mostly not to be reassigning a value uneedlessly, it just felt a bit weird to mutate such value on every re-render, so my take was to limit it with `useEffect`. It is a boolean value, so it won't trigger any rerender indeed, it's more on the case it was updated out of this hook's context. Which thinking now, doesn't happen 😅

Reply:
:-)

We need to be careful with such optimizations, because in this case it's likely making the code less optimized (instead of boolean assignment, we get functions calls, function definitions, etc) but what's even worse is that makes the code less readable.